plugins/zenoh-plugin-trait/src/manager.rs

```diff
@@ -0,0 +1,300 @@
+    PluginsManager<StartArgs, Instance>
+{
+    /// Constructs a new plugin manager with dynamic library loading enabled.
+    pub fn dynamic<S: Into<String>>(loader: LibLoader, default_lib_prefix: S) -> Self {
+        PluginsManager {
+            default_lib_prefix: default_lib_prefix.into(),
+            loader: Some(loader),
+            plugins: Vec::new(),
+        }
+    }
+    /// Constructs a new plugin manager with dynamic library loading disabled.
+    pub fn static_plugins_only() -> Self {
+        PluginsManager {
+            default_lib_prefix: String::new(),
+            loader: None,
+            plugins: Vec::new(),
+        }
+    }
+
+    /// Adds a statically linked plugin to the manager.
+    pub fn declare_static_plugin<
+        P: Plugin<StartArgs = StartArgs, Instance = Instance> + Send + Sync,
+    >(
+        mut self,
+    ) -> Self {
+        let plugin_loader: StaticPlugin<StartArgs, Instance, P> = StaticPlugin::new();
+        self.plugins.push(PluginRecord::new(plugin_loader));
+        log::debug!(
+            "Declared static plugin {}",
+            self.plugins.last().unwrap().name()
+        );
+        self
+    }
+
+    /// Add dynamic plugin to the manager by name, automatically prepending the default library prefix
+    pub fn declare_dynamic_plugin_by_name<S: Into<String>>(
+        &mut self,
+        name: S,
+        plugin_name: &str,
+    ) -> ZResult<&mut dyn DeclaredPlugin<StartArgs, Instance>> {
+        let name = name.into();
+        let plugin_name = format!("{}{}", self.default_lib_prefix, plugin_name);
+        let libloader = self
+            .loader
+            .as_ref()
+            .ok_or("Dynamic plugin loading is disabled")?
+            .clone();
+        log::debug!("Declared dynamic plugin {} by name {}", &name, &plugin_name);
+        let loader =
+            DynamicPlugin::new(name, DynamicPluginSource::ByName((libloader, plugin_name)));
+        self.plugins.push(PluginRecord::new(loader));
+        Ok(self.plugins.last_mut().unwrap())
+    }
+
+    /// Add first available dynamic plugin from the list of paths to the plugin files
+    pub fn declare_dynamic_plugin_by_paths<S: Into<String>, P: AsRef<str> + std::fmt::Debug>(
+        &mut self,
+        name: S,
+        paths: &[P],
+    ) -> ZResult<&mut dyn DeclaredPlugin<StartArgs, Instance>> {
+        let name = name.into();
+        let paths = paths.iter().map(|p| p.as_ref().into()).collect();
+        log::debug!("Declared dynamic plugin {} by paths {:?}", &name, &paths);
+        let loader = DynamicPlugin::new(name, DynamicPluginSource::ByPaths(paths));
+        self.plugins.push(PluginRecord::new(loader));
+        Ok(self.plugins.last_mut().unwrap())
+    }
+
+    fn get_plugin_index(&self, name: &str) -> Option<usize> {
+        self.plugins.iter().position(|p| p.name() == name)
+    }
+
+    /// Lists all plugins
+    pub fn declared_plugins(
```

Comment:
Not a blocker, but I'd rather stick to [Rust naming convention](https://rust-lang.github.io/api-guidelines/naming.html#methods-on-collections-that-produce-iterators-follow-iter-iter_mut-into_iter-c-iter) for methods returning iterators.

Reply:
Done